proposals/infra/INF-0005-Long-Vector-TestPlan.md

```diff
@@ -0,0 +1,243 @@
+assumption.
+
+# Implementation phases
+Do the test work in two simple phases.
+
+1. Implement and validate (locally against WARP) the 3 test categories.
+2. HLK related work:
+     - Add a SM 6.9 HLK requirement. Includes updating the HLK requirements doc.
+     - Update mm_annotate_shader_op_arith_table.py to annotate the new test cases
+        with HLK GUIDS and requirements
+     - Add new tests to HLK playlist
+
+# Shipping
+1. Tests will be shared privately with IHV's along with the latest DXC and
+latest Agility SDK for tesing and validation.
+
+2. The tests will ship with the HLK at a TBD date in a later OS release.
+
+# New HLK Tests
+mm_annotate_shader_op_arith_table.py (WinTools repo) will need to be updated to
+recognize any new tests (not test cases) added. And additional GUIDs added for new
+test cases. mm_annotate_shader_op_arith_table.py is called by mm-hlk-update.py when
+converting from 'ExecTests' to the HLK 'DxilConf' tests. The aforementioned
+*table.py script is run by Integration.HLKTestsUpdate.yaml
+
+# Test Validation Requirements
+The following statements must be true and validated for this work to be considered
+completed.
+ - All new test cases pass when run locally against a WARP device
+ - All new test cases are confirmed to be lit up and available in the HLK when the
+    target device reports support
+ - All new tests/test cases are added to the HLK playlist
+
+# Requirements
+     - Greg's long vector changes. Check-in ETA of 3/7/25.
+     - WARP long vector support (Jesse). Needs Greg's work, or a private WIP branch
+        of Greg's work. ETA of ~1 week to implement.
+
+# Notes
+     - The 'ExecTests' (ExecutionTest.cpp) are used to generate the HLK tests (DxilConf
+        tests)
+     - Private test binaries/collateral will be shared with IHVs for validation
+        purposes. This will enable IHVs to verify long vector functionality without
+        waiting for an OS/HLK release.
+
+# Open Questions
+     - We speculate that the combiniation of test cases for HLSL operators and
+     HLSL intrinsics which map to DXIL ops should get us all, or most of the
+     coverage required for the 'emulated' intrinsics. But we should audit that.
+
+# HLSL Operators
+These operatores should generate LLVM native ops which use vectors.
+
+| Operator Name | Operator | Notes |
+|-----------|--------------|----------|
+Additive and Multiplicative Operators | +, -, *, /, % |
+Array Operator | [i] |
+Assignment Operators | =, +=, -=, *=, /=, %= |
+Binary Casts | C rules for float and int, C rules or HLSL intrinsics for bool |
+Bitwise Operators | ~, <<, >>, &, \|, ^, <<=, >>=, &=, \|=, ^= | Only valid on int and uint vectors
+Boolean Math Operators | & &, ||, ?: |
+Cast Operator | (type) |
+Comma Operator | , |
+Comparison Operators| <, >, ==, !=, <=, >= |
+Prefix or Postfix Operators| ++, -- |
+Structure Operator | . |
+Unary Operators | !, -, + |
+
+# Mappings of HLSL intrinsics to DXIL opcodes or LLVM native operations
```

Comment:
Wondering where you sourced this list of operators from?  Not all lists out there are exhaustive!

Reply:
These were sourced from [HLSL operators](https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-operators 
)

Reply:
D'oh, I said "operators" and I meant "opcodes".  So specifically thinking about all the various intrinsics / DXIL operations that you want to exercise.  This is where some of Farzon's stuff, like https://github.com/farzonl/HLSLDocGenerator, might come in handy.

Reply:
The HLSL intrinsics were sourced from [0026-hlsl-long-vector-type.md](https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-operators#binary-casts )

Reply:
:) I think that doc's worth treating as authoritative - but not that link!  Try clicking it :)